src/goose/brokers/redis/commands.clj

```diff
@@ -142,18 +142,26 @@
   ;; https://github.com/ptaoussanis/carmine/issues/266
   (wcar* conn (car/blmove src dst "RIGHT" "RIGHT" d/redis-long-polling-timeout-sec)))
 
-(defn list-position [conn list element]
-  (wcar* conn (car/lpos list element) "COUNT" 1))
-
-(defn del-from-list-and-enqueue-front [conn list element]
-  (car/atomic
-    conn atomic-lock-attempts
-    (car/multi)
-    (car/lrem list 1 element)
-    (car/rpush list element)))
-
-(defn del-from-list [conn list element]
-  (wcar* conn (car/lrem list 1 element)))
+(defn del-from-list-and-enqueue-front [conn list elements]
+  (->> (wcar* conn :as-pipeline (doseq [element elements]
+                                  (car/multi)
+                                  (car/lrem list 1 element)
+                                  (car/rpush list element)
+                                  (car/exec)))
+       ;; the response is a flattened list containing result of individual commands
+       ;; ex i/p: ["OK" "QUEUED" "QUEUED" [0 1] "OK" "QUEUED" "QUEUED" [0 1]]
+       ;; the final response is partitioned into vectors with 4 results (i.e response for 1 `element`)
```

Comment:
suggestion: can you also explain why we do this transformation. It isn't apparent from this comment

Reply:
I have added the following comment:
>       To ensure consistency with the contract for handling individual element responses(used earlier using car/atomic), the response format for multiple elements is constructed as sequence of these single element responses.